src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,33 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
+  todo: Todo | null;
+  onClose: () => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ todo, onClose }) => {
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    if (todo) {
+      getUser(todo.userId).then(setUser);
+    }
+  }, [todo]);
+
+  const handleCloseModal = () => {
+    onClose();
+    setUser(null);
+  };
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      {true ? (
+      {!user ? (
         <Loader />
```

Comment:
The `Loader` component is displayed when the `user` is not loaded. Ensure this is the intended behavior. If you want to show the loader only when the user data is being fetched, consider adding a loading state to manage this explicitly.